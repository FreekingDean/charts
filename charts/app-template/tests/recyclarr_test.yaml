tests:
  - it: validates my recyclarr
    set:
      stateful: true
      persistence:
        config:
          existingClaim: recyclarr-0
          globalMounts:
            - path: /config
        config-file:
          enabled: true
          type: configMap
          name: recyclarr-configmap
          globalMounts:
            - path: "/config/recyclarr.yml"
              subPath: recyclarr.yml
      image:
        repository: ghcr.io/recyclarr/recyclarr
        tag: edge
      env:
        TZ: America/New_York
        RADARR__API_KEY: "${RADARR__API_KEY}"
        RADARR4K__API_KEY: "${RADARR4K__API_KEY}"
        SONARR__API_KEY: "${SONARR__API_KEY}"
        SONARR4K__API_KEY: "${SONARR4K__API_KEY}"
    asserts:
      - matchSnapshot: {}
