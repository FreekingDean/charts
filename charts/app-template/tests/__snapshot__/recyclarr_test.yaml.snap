validates my recyclarr:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        reloader.stakater.com/auto: "true"
      labels:
        app.kubernetes.io/component: RELEASE-NAME
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        helm.sh/chart: app-template-0.0.4
      name: RELEASE-NAME
    spec:
      replicas: 2
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app.kubernetes.io/component: RELEASE-NAME
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: RELEASE-NAME
      strategy:
        type: Recreate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: RELEASE-NAME
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: RELEASE-NAME
        spec:
          automountServiceAccountToken: true
          containers:
            - env:
                - name: RADARR4K__API_KEY
                  value: ${RADARR4K__API_KEY}
                - name: RADARR__API_KEY
                  value: ${RADARR__API_KEY}
                - name: SONARR4K__API_KEY
                  value: ${SONARR4K__API_KEY}
                - name: SONARR__API_KEY
                  value: ${SONARR__API_KEY}
                - name: TZ
                  value: America/New_York
              image: ghcr.io/recyclarr/recyclarr:edge
              name: RELEASE-NAME
              volumeMounts:
                - mountPath: /config
                  name: config
                - mountPath: /config/recyclarr.yml
                  name: config-file
                  subPath: recyclarr.yml
          dnsPolicy: ClusterFirst
          enableServiceLinks: false
          hostIPC: false
          hostNetwork: false
          hostPID: false
          serviceAccountName: default
          volumes:
            - name: config
              persistentVolumeClaim:
                claimName: recyclarr-0
            - configMap:
                name: recyclarr-configmap
              name: config-file
  2: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        helm.sh/chart: app-template-0.0.4
      name: RELEASE-NAME
    spec:
      ingressClassName: traefik
      rules:
        - host: .deangalvin.dev
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: null
                path: /
                pathType: Prefix
      tls:
        - hosts:
            - .deangalvin.dev
          secretName: deangalvin-dev-production-tls
  3: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        helm.sh/chart: app-template-0.0.4
      name: RELEASE-NAME
    spec:
      endpoints:
        - interval: 1m
          path: /metrics
          port: http
          scheme: http
          scrapeTimeout: 10s
      jobLabel: RELEASE-NAME
      namespaceSelector:
        matchNames:
          - NAMESPACE
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: RELEASE-NAME
          app.kubernetes.io/service: RELEASE-NAME
